### PR TITLE
[1.x] Allow `redirect()->intended()` responses to be resolved via the Container

### PR DESCRIPTION
**Pull request by crynobone**
Signed-off-by: Mior Muhammad Zaki <crynobone@gmail.com>

### CODE REVIEW

**Comment by taylorotwell** [comment]
@crynobone is this fixing a issue filed on GitHub?

File: src/FortifyServiceProvider.php [hunk 1]
```diff
@@ -73,6 +75,10 @@ public function register()
         $this->app->bind(StatefulGuard::class, function () {
             return Auth::guard(config('fortify.guard', null));
         });
+
+        $this->app->bind(PasswordBroker::class, function () {
```
Comment (taylorotwell) [inline src/FortifyServiceProvider.php:79]: Is this change totally necessary? I'm worried about overriding this binding for existing applications.

File: src/Http/Controllers/NewPasswordController.php [hunk 2]
```diff
@@ -87,6 +87,6 @@ function ($user) use ($request) {
      */
     protected function broker(): PasswordBroker
     {
-        return Password::broker(config('fortify.passwords'));
+        return app(PasswordBroker::class);
```
Comment (crynobone) [inline src/Http/Controllers/NewPasswordController.php:90]:
```suggestion
        return Password::broker(config('fortify.passwords'));
```

File: src/Http/Controllers/PasswordResetLinkController.php [hunk 3]
```diff
@@ -54,6 +54,6 @@ public function store(Request $request): Responsable
      */
     protected function broker(): PasswordBroker
     {
-        return Password::broker(config('fortify.passwords'));
+        return app(PasswordBroker::class);
```
Comment (crynobone) [inline src/Http/Controllers/PasswordResetLinkController.php:57]:
```suggestion
        return Password::broker(config('fortify.passwords'));
```

File: src/FortifyServiceProvider.php [hunk 4]
```diff
@@ -2,9 +2,11 @@
 
 namespace Laravel\Fortify;
 
+use Illuminate\Contracts\Auth\PasswordBroker;
 use Illuminate\Contracts\Auth\StatefulGuard;
```
Comment (crynobone) [inline src/FortifyServiceProvider.php:6]:
```suggestion
use Illuminate\Contracts\Auth\StatefulGuard;
```

File: src/FortifyServiceProvider.php [hunk 5]
```diff
@@ -2,9 +2,11 @@
...
 use Illuminate\Contracts\Cache\Repository;
 use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Password;
 use Illuminate\Support\Facades\Route;
```
Comment (crynobone) [inline src/FortifyServiceProvider.php:10]:
```suggestion
use Illuminate\Support\Facades\Route;
```

File: src/FortifyServiceProvider.php [hunk 6]
```diff
@@ -73,6 +75,10 @@ public function register()
         $this->app->bind(StatefulGuard::class, function () {
             return Auth::guard(config('fortify.guard', null));
         });
+
+        $this->app->bind(PasswordBroker::class, function () {
+            return Password::broker(config('fortify.passwords'));
+        });
```
Comment (crynobone) [inline src/FortifyServiceProvider.php:81]:
```suggestion
        });
```

File: src/FortifyServiceProvider.php (hunk 1 above)
Comment (crynobone) [reply src/FortifyServiceProvider.php]: I reverted the password broker changes.

**Pull request merged**